### PR TITLE
Fix hydration

### LINKED ISSUE #1557: Remix hydration issues with the builder and published saas apps
The issue is caused by some extensions like 
[Dark Mode](https://chrome.google.com/webstore/detail/dark-mode/dmghijelimhndkbmpgbldicpogfkceaj/related)
On the https://alpha.webstudio.is/dashboard page
(and any published site https://webstudio.is, but on published sites, even hydration is broken we are still able to see content)

The extension adds few style tags and breaks hydration. 
(Saas is still renderable but has a console error about that hydration was broken)

Description and proposals of how to solve here 
https://github.com/kiliman/remix-hydration-fix
i.e. using https://github.com/Xiphe/remix-island

### PR DESCRIPTION
**Pull request by istarkov**
## Description

closes #1557

## Steps for reproduction

Install https://chrome.google.com/webstore/detail/dark-mode/dmghijelimhndkbmpgbldicpogfkceaj plugin
See our website doesn't work but current PR is

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/builder/builder.tsx [hunk 1]
```diff
@@ -58,12 +54,9 @@ import {
 
 registerContainers();
 
+// Can cause FOUC because of remix-island, be very accurate adding anything here
 export const links = () => {
-  return [
-    { rel: "stylesheet", href: interFont },
-    { rel: "stylesheet", href: robotoMonoFont },
```
Comment (kof) [inline apps/builder/app/builder/builder.tsx:64]: where did this roboto font go?

File: apps/builder/app/shared/remix/handle-request.server.tsx [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
 import { RemixServer } from "@remix-run/react";
 import type { EntryContext } from "@remix-run/node";
-import { insertCriticalCss } from "./insert-critical-css";
+import { flushCss } from "@webstudio-is/design-system";
+import { renderHeadToString } from "remix-island";
+import { Head } from "./root";
+// eslint-disable-next-line import/no-internal-modules
+import interFont from "@fontsource/inter/variable.css";
+// eslint-disable-next-line import/no-internal-modules
+import manropeVariableFont from "@fontsource/manrope/variable.css";
```
Comment (kof) [inline apps/builder/app/shared/remix/handle-request.server.tsx:10]: bad part about this structure is that now you have a central place for all fonts, before that builder, dashboard and login was managing their fonts

e.g. dashboard and login uses brand-defined stuff, where builder doesn't

Comment (istarkov) [reply apps/builder/app/shared/remix/handle-request.server.tsx:10]: The good part that no fouc caused by remix-island

Comment (istarkov) [reply apps/builder/app/shared/remix/handle-request.server.tsx:10]: Also I wrote a comment
https://github.com/webstudio-is/webstudio-builder/pull/1621/files#diff-2f32780da9f3d635a9c0b765d6cb7888c66e2db35f0572f3ea0971751a0814c1R28-R30

File: apps/builder/app/shared/remix/handle-request.server.tsx [hunk 3]
```diff
@@ -13,10 +22,33 @@ export const handleRequest = (
...
+
   responseHeaders.set("Content-Type", "text/html");
   responseHeaders.set("Accept-CH", "Sec-CH-Prefers-Color-Scheme");
 
-  return new Response("<!DOCTYPE html>" + insertCriticalCss(markup), {
+  // data-builder="true" to distinguish between builder and canvas at the entry.client.tsx
+  // inject fonts and base body styles here instead of `export links = ` to avoid FOUC
+  const body = `
+    <!DOCTYPE html>
+    <html lang="en" data-builder="true">
```
Comment (kof) [inline apps/builder/app/shared/remix/handle-request.server.tsx:34]: I feel like this is all leaky abstraction, we shouldn't have app specific stuff in the layout we are reusing across different apps

Comment (kof) [reply apps/builder/app/shared/remix/handle-request.server.tsx:34]: effectively you are writing the app name inside the layout

Comment (kof) [reply apps/builder/app/shared/remix/handle-request.server.tsx:34]: could be data-app="builder" and consistent, so all apps would be there, but even then, its still not great, I bet we can avoid it all

**Review by kof — APPROVED** [review]

**Review by kof — APPROVED** [review]

**Pull request merged**